src/app/users/users.component.ts

```diff
@@ -0,0 +1,62 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {User} from './user.model';
+import {from, Observable, Subscription} from 'rxjs';
+import {filter} from 'rxjs/operators';
+
+@Component({
+  selector: 'app-users',
+  templateUrl: './users.component.html',
+  styleUrls: ['./users.component.less']
+})
+export class UsersComponent implements OnInit {
+  usersArray: User[] = [];
+  primaryUserArray: User[] = [];
+  filteredUserArray: User[] = [];
+  isFiltered = false;
+  private subscription: Subscription;
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+    this.usersArray = this.generateArray();
+  }
+
+  generateArray(): User[] {
+    const firstNamesArray = ['John', 'Alexa', 'Max', 'Vika', 'Andrew'];
+    const lastNamesArray = ['Trevino', 'Hamilton', 'Newman', 'Henderson', 'Erickson'];
```

Comment:
Move it to common constant file

Reply:
Created constant file and moved arrays into it.